### PR TITLE
Add openshift/cluster-api-provider-ibmcloud repo

### PR DESCRIPTION
**Pull request by srcarrier**

### CODE REVIEW

**Comment by JoelSpeed** [comment]
This looks good apart from the conflict and the fact that the IBM Cloud E2E jobs aren't merged yet

**Comment by srcarrier** [comment]
> This looks good apart from the conflict and the fact that the IBM Cloud E2E jobs aren't merged yet

@JoelSpeed 
1. conflict resolved
2. should I reference the aws e2e jobs in the interim, until the ibm e2e jobs get merged?

**Comment by JoelSpeed** [comment]
> should I reference the aws e2e jobs in the interim, until the ibm e2e jobs get merged?

That could be a way forward just to get things running.

Seems @cjschaef is working on getting the IBM side of things up and running, perhaps they can weigh in here with a rough eta for the IBM E2E workflows being ready?

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:6]: just curious, do we need the 4.6 builders?

**Review by elmiko — COMMENTED** [review]
quick question about versions

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml
Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml:90]: given the discussion about using the aws-e2e to get things going, should these be trimmed back for now? (eg replaced with just the aws-e2e sanity check)

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (srcarrier) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:6]: idk - I was mimicking the content of the other provider docs.
Shall I remove lines 6-9?

**Comment by cjschaef** [comment]
@JoelSpeed Sorry, I've been off task for the last month. But the IBMCloud e2e test setup is waiting for the last of the initial installer PR's to get merged. I think only one or two left at this point.

I also have some additional changes to make to the e2e setup as well, but have not been able to do so due to other duties. Hopefully once the installer is ready though, the e2e PR's can start rolling in.

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml
Comment (srcarrier) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml:90]: Replace the code referenced above with the following ... ?
```
- as: e2e-aws
  steps:
    cluster_profile: aws
    workflow: openshift-e2e-aws
```

Comment (cjschaef) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml:90]: I think so. We won't have any `upgrade` or `serial` tests to start with for ibmcloud e2e.
I think just the `e2e-aws` tests would be sufficient until we can add and expand ibmcloud CI workflows

Comment (elmiko) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml]: agreed, i think this is fine for the first pass just to make sure we aren't disrupting the status quo. then we can layer the ibm cloud specific tests on as we create them and get the infra setup.

Comment (srcarrier) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9.yaml]: done

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (elmiko) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:6]: i don't think we do, since there will be no ibm cloud in 4.6 afaik. cc @JoelSpeed

**Comment by srcarrier** [comment]
@elmiko @JoelSpeed any further code change requests?
Are all of the checks expected to be green at this point or is there a specific subset I need to address prior to merge?
Thanks in advance

**Comment by elmiko** [comment]
looks like we need to fix up a few more things to get these required tests passing, for example we need to add an OWNERS file to this directory. i'm not sure about the other required tests, need to do a little digging.

**Comment by elmiko** [comment]
ok, looking through the failures on the required tests seems like we need a few things

it seems like most of the tests are failing because they don't recognize `ibmcloud` as a valid cluster profile. i'm not sure if this is referring to the CI target, but i'm guessing we still need to modify a component to allow `ibmcloud`. these tests are all failing because of that:
* ci-operator-config
* ci-operator-config-metadata 
* ci-operator-registry
* correctly-sharded-config
* generated-config

the owners test is failing because we need to drop an OWNERS file in this commit. (you added this as i was typing XD)

**Comment by srcarrier** [comment]
@elmiko @JoelSpeed wrt the _"invalid cluster profile"_ error, do we need `ibmcloud` included in the following case stmt?
https://github.com/openshift/ci-operator/blob/d091d71bb3915fb26305571cb258dbee2acb32ae/pkg/api/config.go#L181-L187

**Comment by elmiko** [comment]
> @elmiko @JoelSpeed wrt the _"invalid cluster profile"_ error, do we need `ibmcloud` included in the following case stmt?
> https://github.com/openshift/ci-operator/blob/d091d71bb3915fb26305571cb258dbee2acb32ae/pkg/api/config.go#L181-L187

that makes sense to me, i'm just not sure if it's trying to validate the provider type for testing cluster that is created. could use a little more advice here.

**Comment by elmiko** [comment]
i think part of the problem might also be that we have `ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml` referencing the full tests on an ibmcloud. we can't use aws there, but i wonder if we need to remove those steps as well and just have the e2e-aws in there.

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:81]: i'm kinda wondering if we need to replace this with the e2e-aws stuff from the release-4.9.yaml, at least until we have the ibmcloud ready for testing.

Comment (srcarrier) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml]: done
@elmiko any suggested edits for `e2e-ibmcloud` references within the following presubmit yamls?
Should I convert them over to `e2e-aws` or ?
```
/ci-operator/jobs/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main-presubmits.yaml
/ci-operator/jobs/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-release-4.9-presubmits.yaml
```

Comment (elmiko) [reply ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml]: i think so (sorry if i missed those), imo i think we should just make this PR do only the e2e-aws tests across the board. lets just make it simple, then when the ibmcloud is ready as a platform we can add on the other tests. this way we can at least get the basic _smoke test_ working and then proceed.

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:17]: This should match the go module name, which is `github.com/openshift/cluster-api-provider-ibmcloud`

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/openshift-cluster-api-provider-ibmcloud-main.yaml:19]: This file doesn't exist in the repository? Do you mean `Dockerfile`?

**Review by JoelSpeed — COMMENTED** [review]
Please fix up the two comments below and run `make jobs` to update the generated jobs, then I think this looks correct syntactically at least

**Comment by elmiko** [comment]
we are currently blocking on this https://github.com/openshift/cluster-api-provider-ibmcloud/pull/5

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/OWNERS
Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/OWNERS:6]: we should remove danil

File: ci-operator/config/openshift/cluster-api-provider-ibmcloud/OWNERS
Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/OWNERS:18]: we should remove `danil-grigorev`, `enxebre`, and `michaelgugino` from these lists

Comment (elmiko) [inline ci-operator/config/openshift/cluster-api-provider-ibmcloud/OWNERS:18]: we should remove `danil-grigorev`, `enxebre`, and `michaelgugino` from these lists, and add `lobziik`.

**Review by elmiko — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/cluster-api-provider-ibmcloud/OWNERS
Comment (elmiko) [inline ci-operator/jobs/openshift/cluster-api-provider-ibmcloud/OWNERS:18]: we should remove `danil-grigorev`, `enxebre`, and `michaelgugino` from these lists, and add `lobziik`.

**Review by elmiko — CHANGES_REQUESTED** [review]

**Review by elmiko — COMMENTED** [review]
/lgtm

**Review by alvaroaleman — APPROVED** [review]
/label tide/merge-method-squash

**Pull request merged**
